Unity-Technologies/ScriptableRenderPipeline
# [Shader Graph] Fix Position Node upgrader

### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing? 
Fixed an issue where new Position nodes added to graphs would always be "upgraded", causing the coordinate space drop down to switch from World to Absolute World when entering play mode or restarting the project. 

[public link](https://issuetracker.unity3d.com/product/unity/issues/guid/1184617/)
[internal link](https://fogbugz.unity3d.com/f/cases/1184617/)

---
### Testing status

**Manual Tests**: What did you do?
- Other: 
Previous repro behavior: 
- Create a new graph, add a Position node. leave the coordinate space on default World. Save graph.
- Enter and exit playmode to force shader compilation. 
- Open the graph, observe that World has been automatically set to Absolute World. 

Fix behavior: 
- Create a new graph, add a Position node. leave the coordinate space on default World. Save graph.
- Enter and exit playmode to force shader compilation. 
- Open the graph, observe that the coordinate space has not changed.  

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.
This was initially intended to be an upgrader for the position node when Absolute World space was added. Because HDRP uses different values, the intent was that any existing graphs would maintain the same output even though the entry changed. The node version was never set properly and it caused the Position node to always run through the upgrader no matter what. 
Any graphs with the Position node already in use should not see any change in behavior. This only applies to newly created Position nodes after this fix is merged.
